js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -2272,6 +2306,12 @@ const EditingController = modules.ViewController.inherit((function() {
 
             if(editDataIndex < 0) {
                 editDataIndex = changes.length;
+
+                const internalData = this._getInternalData(options.key, true);
+                internalData.oldData = options.oldData;
```

Comment:
Add the _addInternalData method and use it here. Also the second boolean parameter is not used in _getInternalData.